libparsec/crates/zstd/src/dirty.rs

```diff
@@ -0,0 +1,157 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::io;
+
+/// Decompress from the given source as if using a `Decoder`.
+///
+/// The input data must be in the zstd frame format.
+pub fn decode_all<R: io::Read>(source: R) -> io::Result<Vec<u8>> {
+    let mut decoder = ruzstd::StreamingDecoder::new(source).map_err(io::Error::other)?;
+    let mut result = Vec::new();
+
+    use std::io::Read;
+    decoder.read_to_end(&mut result)?;
+
+    Ok(result)
+}
+
+/// Compress all data from the given source as if using an `Encoder`.
+///
+/// Compressed data will be appended to `destination`.
+///
+/// A level of `0` uses zstd's default (currently `3`).
```

Comment:
You mean currently 3 in parsec overall ?

Reply:
I've copied this comment from zstd's documentation :eyes:

Reply:
I've updated the comment to:
```rust
/// Compress all data from the given source as if using an `Encoder`.
///
/// Compressed data will be appended to `destination`.
///
/// Note `level` is simply ignored here given this implementation of compressor
/// doesn't do any actual compression !
```